src/main/java/parking/VehicleParking.java

```diff
@@ -12,30 +16,59 @@
     public static Map<String, AbstractVehicle> numberVehicleMap = new HashMap<>();
     public static Map<String, AbstractVehicle> sortedMap = new HashMap<>();
 
-    public static void printVehicleList(List list) {
+    // Task #1 Add possibility for user to retrieve list of vehicles on a parking lot
+    public static void printList(List list) {
         list.stream().forEach(System.out::println);
     }
 
-    public static void filterByNumber(String template) {
-        for (AbstractVehicle vehicle : vehicleList) {
-            if (vehicle.getNumber().substring(0, 2).equals(template)) {
-                System.out.println(vehicle);
-            }
-        }
+    //Task #2 Add possibility for user to retrieve list of vehicles filtered by type (e.g. car, bike, lorry etc)
+    public static void filterByType(String vehicleType) {
+        vehicleList.stream().filter(o -> o.getVehicleType().equals(vehicleType)).forEach(System.out::println);
     }
 
+    // Task #3 Sort vehicles list according to duration it is staying on a parking lot
+    public static void sortByDurations() {
+        // vehicleList.stream().sorted((o1, o2) -> new DateComparator()).forEach(System.out::println);
+        List<AbstractVehicle> sortedUsers = vehicleList.stream()
+                .sorted(Comparator.comparing(AbstractVehicle::getDateIn).reversed())
+                .collect(Collectors.toList());
+        printList(sortedUsers);
+    }
 
-    public static boolean isNumberExist(String template) {
-        for (AbstractVehicle vehicle : vehicleList) {
-            if (vehicle.getNumber().substring(0, 2).equals(template)) {
-                return true;
-            }
+    /* Task #4   Check if at least one vehicle in your list corresponds to some
+     search criteria (e.g. vehicle is registered in Lviv (plate number starts with ‘BC’))
+     */
+    public static boolean isNumberExist(String number) {
+        return vehicleList.stream().filter(o -> o.getNumber().substring(0, 2).equals(number)).findFirst().isPresent();
+    }
+
+    // Task #5    Check if all vehicles correspond to some search criteria (e.g. name of Vehicle owner is Ivan)
+    public static boolean isAllOwners(String ownerSearch) {
+        return vehicleList.stream().allMatch(o -> o.getOwner().equals(ownerSearch));
+    }
+
+    // Task #6  Check if none of the vehicles from list corresponds to some search criteria (e.g. vehicle type is motorcycle)
+    public static boolean isAllTypes(String typeSearch) {
+        return (vehicleList.stream().noneMatch(o -> o.getVehicleType().equals(typeSearch)));
+    }
+
+    /* Task #7  Traverse vehicle list, collect all unique registration numbers,
+    sort them alphabetically and print them to console using comma as a delimiter
+     */
+    public static String getCarByUniqueNumber() {
+
+        sortedNumbersList = vehicleList.stream().map(vehicleList -> vehicleList.getNumber()).distinct().sorted().collect(Collectors.toList());
```

Comment:
lambda vehicleList -> vehicleList.getNumber() can be replaced with method reference

Reply:
done